doc/cephadm/upgrade.rst

```diff
@@ -188,3 +188,101 @@ you need. For example, the following command upgrades to a development build:
   ceph orch upgrade start --image quay.io/ceph-ci/ceph:recent-git-branch-name
 
 For more information about available container images, see :ref:`containers`.
+
+Staggered Upgrade
+=================
+
+Some users may prefer to have their upgrade be done in shorter bursts rather
+than one fire-and-forget upgrade command. The upgrade command, starting
+in 16.2.9 and 17.2.1 allows specifying parameters to limit which daemons are
+upgraded by a single upgrade command. The options in include ``daemon_types``,
+``services`` and ``hosts``. ``daemon_types`` takes a comma separated list
+of daemon types and will only upgrade daemons of those types. ``services``
+is mutually exclusive with ``daemon_types``, only takes services of one type
+at a time (e.g. can't provide an osd and rgw service at the same time) and
+will only upgrade daemons belonging to those services. ``hosts`` can be combined
+with ``daemon_types`` or ``services`` or provided on its own. The argument
+follows the same format as the command line options for :ref:`orchestrator-cli-placement-spec`.
+
+Example, specifying daemon types and hosts:
+
+.. prompt:: bash #
+
+  ceph orch upgrade start --image <image-name> --daemon-types mgr,mon --hosts host1,host2
+
+Example, specifying services:
+
+.. prompt:: bash #
+
+  ceph orch upgrade start --image <image-name> --services rgw.example1,rgw.example2
+
+.. note::
+
+   cephadm strictly enforces an order to the upgrade of daemons that is still present
+   in staggered upgrade scenarios. The current upgrade ordering is
+   ``mgr -> mon -> crash -> osd -> mds -> rgw -> rbd-mirror -> cephfs-mirror -> iscsi -> nfs``.
+   If you attempt to specify parameters that would force us to upgrade daemons out of
+   order, the upgrade command will block and inform you of which daemons earlier
+   in the upgrade order will be missed using your parameters.
+
+.. note::
+
+   In order to verify the upgrade order will not be broken, upgrade commands with
+   extra limiting parameters must first validate the options before beginning the
+   upgrade. This may require pulling the image you are trying to upgrade to. Do
+   not be surprised if the upgrade start command takes a while to return when limiting
+   parameters are provided.
+
```

Comment:
Suggest:
>Upgrade commands with
   limiting parameters will validate the options before beginning the
   upgrade, which may require pulling the new container image. Do
   not be surprised if the upgrade start command takes a while to return when limiting
   parameters are provided.